### ISSUE TITLE
CSS/React Tree Persistence in Gatsby v2

### ISSUE THREAD

**Issue by daneden** [gatsbyjs/gatsby#6956]
### TL;DR
How can we ensure that shared resources such as CSS/fonts, which I previously would have just thrown in `layout.js`, are requested only once in Gatsby V2? Following the migration guide, I end up re-downloading `import`-ed CSS/fonts with every followed `Link`.

***

I’ve been [working on](https://github.com/daneden/daneden.me/issues/27) migrating my [website](https://daneden.me)  to Gatsby v2, with peaks of elated success and valleys of dismal defeat. I'm here about one of those valleys.

I gathered in the migration guide that the changes to Gatsby's layout mean that pages are re-rendered in full, rather than the old way of doing things:

```
// old world
<Layout>
  {/* only children() would be re-rendered on page change */}
  {children()}
</Layout>

// new world
// everything here re-renders on page change
<Layout>
  {children}
</Layout>
```

This makes sense, as far as I can tell, except it's introduced one big snag for my site. I import my CSS (and by extension, some web fonts) in my `Layout.js`, which means that on each page load, a new request is made for the same CSS.

**Worse still**, using `Link` means that some pre-fetching work is done, and because the whole tree is re-rendered, I end up with the same CSS being called for *every single `Link`-ed page*.

You can actually experience this on one of [my preview deploys](https://5b6281c9792f893573d93e8e--de-home.netlify.com/blog). Open that page and you'll experience the page flickering as it grabs the CSS and fonts multiple times. If you check the network resources, you'll also experience over 100 requests and >2MB of downloads. Compare this to my [current site](https://daneden.me/blog/), which weighs in at 31 requests for each of the `Link`-ed pages JS, and only one download of the CSS/fonts.

Long-winded way to say, how can we ensure that shared resources such as CSS/fonts, which I previously would have just thrown in `layout.js`, are requested only once in Gatsby V2?

**Comment by daneden** [comment]
Never mind! I was kindly [directed on Twitter](https://twitter.com/chase_mccoy/status/1024871636590317573) by @chasemccoy to use `gatsby-browser.js`, which I didn’t realize was a thing.
